### PR TITLE
Add reducers

### PR DESCRIPTION
**Pull request by niicode**
In this milestone, I was able to achieve the following

- Added store to manage state
- Added book and categories reducers

### CODE REVIEW

File: src/redux/books/book.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+export const ADD_BOOK = 'ADD_BOOK';
```
Comment (fobadara) [inline src/redux/books/book.js]: [REQUIRED] Kindly follow the docs pattern in your action types value. It should be like this `bookstore/books/ADD_BOOK`
[REQUIRED] You should be exporting action container methods, not the action type variable

File: src/redux/books/book.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+export const REMOVE_BOOK = 'REMOVE_BOOK';
```
Comment (fobadara) [inline src/redux/books/book.js]: [REQUIRED] Kindly follow the docs pattern in your action types value. It should be like this `bookstore/books/REMOVE_BOOK`
[REQUIRED] You should be exporting action container methods, not the action type variable

File: src/redux/books/book.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+export const EDIT_BOOK = 'EDIT_BOOK';
```
Comment (fobadara) [inline src/redux/books/book.js]: [OPTIONAL] Kindly follow the docs pattern in your action types value. It should be like this `bookstore/books/EDIT_BOOK`

File: src/redux/books/book.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    },
+    {
+      id: 3,
+      title: 'Oh Joy',
+      author: 'Neil Amponsah',
+      category: 'Romance',
+      progress: 20,
+    },
+  ],
+};
```
Comment (fobadara) [inline src/redux/books/book.js]: [REQUIRED] Your initial state should be an empty array not an object with an array that contains items. Kindly fix that. You can refer back to the project instructions.

File: src/redux/books/book.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  };
+}
+
+export function editBook(id, book) {
+  return {
+    type: EDIT_BOOK,
+    id,
+    book,
+  };
+}
```
Comment (fobadara) [inline src/redux/books/book.js]: [REQUIRED] You implemented a personal feature but forgot about a requirement. Kindly create the `removeBook` action creator and add it to the reducer

File: src/redux/categories/categories.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+export const CHECK_STATUS = 'CHECK_STATUS';
```
Comment (fobadara) [inline src/redux/categories/categories.js]: [REQUIRED] Kindly follow the duck's pattern for the value of this action type

File: src/redux/categories/categories.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+export const initialState = {
+  status: [],
+};
```
Comment (fobadara) [inline src/redux/categories/categories.js]: [REQUIRED] Kindly make this an empty array, not an object

**Review by fobadara — CHANGES_REQUESTED** [review]
# REQUIRES CHANGES ♻️ 
Hi  @niicode, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Descriptive README and PR
✅ Linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider:
- [OPTIONAL] Good job writing a commit message that follows some of the best practices but it is also important for you t commit more often. It will help your colleagues understand your code properly and could also save you from headaches like reverting changes in the future.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categories.js (hunk 7 above)
Comment (niicode) [reply src/redux/categories/categories.js]: Hi @fobadara, can you give a little explanation of what you mean by this should be an array, not an object.

**Review by Bondok6 — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

## Hi @niicode 🙋‍♂️, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Great job implementing the requirements by the previous reviewer ✅
- Good job using store, actions, and reducers. ✅
- Your project is professional ✅
- Your repo is professional ✅
- Professional README.md file 💯


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**